warehouse/upload.go

```diff
@@ -824,19 +832,78 @@ func (job *UploadJobT) updateTableSchema(tName string, tableSchemaDiff warehouse
 		return nil
 	}
 
-	if err := job.addColumnsToWarehouse(tName, tableSchemaDiff.ColumnMap); err != nil {
-		return err
+	if err = job.addColumnsToWarehouse(tName, tableSchemaDiff.ColumnMap); err != nil {
+		return fmt.Errorf("adding columns to warehouse: %w", err)
+	}
+
+	if err = job.alterColumnsToWarehouse(tName, tableSchemaDiff.AlteredColumnMap); err != nil {
+		return fmt.Errorf("altering columns to warehouse: %w", err)
 	}
 
-	for _, columnName := range tableSchemaDiff.StringColumnsToBeAlteredToText {
-		err = job.whManager.AlterColumn(tName, columnName, "text")
+	return nil
+}
+
+func (job *UploadJobT) alterColumnsToWarehouse(tName string, columnsMap map[string]string) error {
+	var responseToAlerta []model.AlterTableResponse
+	var errs []error
+
+	for columnName, columnType := range columnsMap {
+		res, err := job.whManager.AlterColumn(tName, columnName, columnType)
 		if err != nil {
-			pkgLogger.Errorf("Altering column %s in table: %s.%s failed. Error: %v", columnName, job.warehouse.Namespace, tName, err)
-			break
+			errs = append(errs, err)
+			continue
 		}
+
+		if res.IsDependent {
```

Comment:
What does this check mean here ?

Reply:
```Go
type AlterTableResponse struct {
	IsDependent bool // true if the column is dependent on another view or rules, false otherwise
	Query       string
}
```

We have added this component which is being used when we do the alter. This is being set to true, if the dropping column is dependent on another view or rules.